src/gt4py/next/type_system/type_translation.py

```diff
@@ -28,6 +28,16 @@
 from gt4py.next.type_system import type_info, type_specifications as ts
 
 
+class SizeArg:
```

Comment:
This is a compile-time standin for a dimension size argument (as used in the implicit domain case during AOT transforms). For reviewers: let me know if you think this should move to `next.otf.arguments`.